### ISSUE TITLE
wallabag iOS app crashes after successful (?) registration

### ISSUE THREAD

**Issue by crendl** [wallabag/ios-app#148]
| Q             | A
| ------------- | ---
| Device?       | iPhone
| Device Os version    | 10.2.1
| App Version   | 2.3.3
| Wallabag server version | self-hosted

### Steps to reproduce/test case

1. Download wallabag v2 iOS app from the store
2. Run the app, enter server info, ClientID and -Secret, user and password
3. App crashes once the registration steps are finished
4. Starting the app a second time results in an immediate crash. 
Nevertheless, the notifications pop-up appears asking to allow notifications from wallabag.
5. Starting the app a third time results in an immediate crash.

### crash log

```
Process:             wallabag [38873]
Path:                /private/var/containers/Bundle/Application/4E71DE0A-49DA-4342-A4EC-DB137CEFF126/wallabag.app/wallabag
Identifier:          fr.district-web.wallabag
Version:             114 (2.3.3)
Code Type:           ARM-64 (Native)
Role:                Foreground
Parent Process:      launchd [1]
Coalition:           fr.district-web.wallabag [3078]


Date/Time:           2018-07-24 10:58:40.5392 +0200
Launch Time:         2018-07-24 10:58:39.0544 +0200
OS Version:          iPhone OS 10.2.1 (14D27)
Report Version:      104

Exception Type:  EXC_BREAKPOINT (SIGTRAP)
Exception Codes: 0x0000000000000001, 0x000000010010ece4
Termination Signal: Trace/BPT trap: 5
Termination Reason: Namespace SIGNAL, Code 0x5
Terminating Process: exc handler [0]
Triggered by Thread:  0

Filtered syslog:
None found

Thread 0 name:  Dispatch queue: com.apple.main-thread
Thread 0 Crashed:
0   wallabag                      	0x000000010010ece4 0x1000fc000 + 77028
1   UIKit                         	0x0000000199692ee0 0x19933b000 + 3505888
2   UIKit                         	0x0000000199429148 0x19933b000 + 975176
3   UIKit                         	0x00000001994288c8 0x19933b000 + 973000
4   UIKit                         	0x0000000199428654 0x19933b000 + 972372
5   UIKit                         	0x0000000199428488 0x19933b000 + 971912
6   UIKit                         	0x00000001994b8f20 0x19933b000 + 1564448
7   wallabag                      	0x000000010010e0c8 0x1000fc000 + 73928
8   wallabag                      	0x000000010010ec5c 0x1000fc000 + 76892
9   UIKit                         	0x000000019934a924 0x19933b000 + 63780
10  UIKit                         	0x00000001994042bc 0x19933b000 + 823996
11  UIKit                         	0x0000000199404194 0x19933b000 + 823700
12  UIKit                         	0x00000001994034dc 0x19933b000 + 820444
13  UIKit                         	0x0000000199402f28 0x19933b000 + 818984
14  UIKit                         	0x0000000199402adc 0x19933b000 + 817884
15  UIKit                         	0x0000000199402a40 0x19933b000 + 817728
16  UIKit                         	0x0000000199347a80 0x19933b000 + 51840
17  QuartzCore                    	0x00000001967f59d8 0x1966e0000 + 1137112
18  QuartzCore                    	0x00000001967ea4cc 0x1966e0000 + 1090764
19  QuartzCore                    	0x00000001967ea38c 0x1966e0000 + 1090444
20  QuartzCore                    	0x00000001967673e0 0x1966e0000 + 553952
21  QuartzCore                    	0x000000019678ea68 0x1966e0000 + 715368
22  QuartzCore                    	0x000000019678f488 0x1966e0000 + 717960
23  CoreFoundation                	0x00000001934420c0 0x193365000 + 905408
24  CoreFoundation                	0x000000019343fcf0 0x193365000 + 896240
25  CoreFoundation                	0x000000019336e2d8 0x193365000 + 37592
26  UIKit                         	0x00000001993b57b0 0x19933b000 + 501680
27  UIKit                         	0x00000001993b0534 0x19933b000 + 480564
28  wallabag                      	0x0000000100102dc4 0x1000fc000 + 28100
29  libdyld.dylib                 	0x00000001923515b8 0x19234d000 + 17848
```

**Comment by bourvill** [comment]
2.3.4 is for RGPD parameters only as the release note specifies it.

I will try to work on your problem soon

**Comment by bourvill** [comment]
if you are crazy and would like test a next release a beta is available : https://testflight.apple.com/join/RF9hzKHt

**Comment by crendl** [comment]
ciao maxime!

everything is working! after entering my credentials, the app
closed/crashed initially, but after starting it again it worked properly.
also the syncing worked for me.

thank you for your effort!
regards,
clemens

**Comment by crendl** [comment]
forgot to mention, i updated to ios 12.1 previously.

**Comment by giacomocariello** [comment]
I've just installed the testflight version and "on premise" wallabag installation. I can report the same behavior as @crendl : a crash after registration, but then no more crashes (the progress bar keeps going), while official version used to crash after a few secs.

**Comment by bourvill** [comment]
Thank for your feeback!

@giacomocariello How many items do you have? Synchronization ends?

**Comment by giacomocariello** [comment]
1418 items, imported from Pocket with some difficulty (memory consumption limits, fastcgi timeout, etc).
With the iOS app, this is roughly the course of events:

1. I logged in the first time;
2. the progress bar completed synchronization and search worked;
3. opening the app at a later time started a new synchronization, but it hang at the beginning and after a few seconds the app crashed (screen blanked and returned to iOS desktop);
4. reopening the app caused the same as number 3. Tried to figure out if it was an error on the backend, but the actual query received a 200 as response, with about 500K payload;
5. I found this thread and decided to try the beta;
6. after deinstalling, installing the testflight beta, re-logging in, the application crashed again just after logon. After that, it didn't crash again.
7. After adding a new item to the backend, the synchronization didn't complete (spinner kept running indefinitely) and couldn't force it to run again;
8. Killing the app and reopening produced a new sync action, which completed successfully (new item appeared on the lists).
